code/game/objects/effects/effect_system/effects_smoke.dm

```diff
@@ -16,61 +16,64 @@
 	layer = FLY_LAYER
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	var/steps = 0
-	var/lifetime = 5
+	var/lifetime = 10 SECONDS_TO_LIFE_CYCLES
 	var/direction
+	var/causes_coughing = FALSE
 
 /obj/effect/particle_effect/smoke/proc/fade_out(frames = 16)
-	if(alpha == 0) //Handle already transparent case
-		return
-	if(frames == 0)
-		frames = 1 //We will just assume that by 0 frames, the coder meant "during one frame".
-	var/step = alpha / frames
-	for(var/i = 0, i < frames, i++)
-		alpha -= step
-		if(alpha < 160)
-			set_opacity(0)
-		stoplag()
-
-/obj/effect/particle_effect/smoke/New()
+	animate(src, 2 SECONDS, alpha = 0, easing = EASE_IN | CIRCULAR_EASING)
+
+/obj/effect/particle_effect/smoke/New(loc, contains_chemicals = FALSE)
 	..()
 	START_PROCESSING(SSobj, src)
-	lifetime += rand(-1,1)
+	RegisterSignal(src, list(COMSIG_MOVABLE_CROSSED, COMSIG_CROSSED_MOVABLE), PROC_REF(smoke_mob)) //If someone crosses the smoke or the smoke crosses someone
+	GLOB.smokes_active++
+	lifetime += rand(-1, 1)
+	if(contains_chemicals)
+		create_reagents(10)
 
 /obj/effect/particle_effect/smoke/Destroy()
 	STOP_PROCESSING(SSobj, src)
+	UnregisterSignal(src, list(COMSIG_MOVABLE_CROSSED, COMSIG_CROSSED_MOVABLE))
+	GLOB.smokes_active--
 	return ..()
 
 /obj/effect/particle_effect/smoke/proc/kill_smoke()
 	STOP_PROCESSING(SSobj, src)
 	INVOKE_ASYNC(src, PROC_REF(fade_out))
-	QDEL_IN(src, 10)
+	QDEL_IN(src, 2 SECONDS)
 
 /obj/effect/particle_effect/smoke/process()
 	lifetime--
 	if(lifetime < 1)
 		kill_smoke()
-		return 0
+		return FALSE
 	if(steps >= 1)
 		step(src,direction)
 		steps--
-	return 1
-
-/obj/effect/particle_effect/smoke/Crossed(mob/living/M, oldloc)
-	if(!istype(M))
-		return
-	smoke_mob(M)
+	for(var/mob/living/carbon/M in range(1, src))
+		smoke_mob(M)
+	return TRUE
 
-/obj/effect/particle_effect/smoke/proc/smoke_mob(mob/living/carbon/C)
-	if(!istype(C))
+/obj/effect/particle_effect/smoke/proc/smoke_mob(datum/source, mob/living/carbon/breather)
+	SIGNAL_HANDLER //COMSIG_MOVABLE_CROSSED and COMSIG_CROSSED_MOVABLE
+	if(!istype(breather))
 		return FALSE
-	if(lifetime<1)
+	if(lifetime < 1)
 		return FALSE
-	if(!C.can_breathe_gas())
+	if(!breather.can_breathe_gas())
 		return FALSE
-	if(C.smoke_delay)
+	if(breather.smoke_delay)
+		addtimer(CALLBACK(src, PROC_REF(remove_smoke_delay), breather), 10) //Sometimes during testing I'd somehow end up with a permanent smoke delay, so this is in case of that
```

Comment:
```suggestion
		addtimer(CALLBACK(src, PROC_REF(remove_smoke_delay), breather), 1 SECONDS) //Sometimes during testing I'd somehow end up with a permanent smoke delay, so this is in case of that
```